libs/utils/utils.js

```diff
@@ -1391,38 +1391,82 @@ export function enablePersonalizationV2() {
   const enablePersV2 = getMepEnablement('personalization-v2');
   return !!enablePersV2 && isSignedOut();
 }
+export function getCookie(key) {
+  const cookie = document.cookie.split(';')
+    .map((x) => decodeURIComponent(x.trim()).split(/=(.*)/s))
+    .find(([k]) => k === key);
+  return cookie ? cookie[1] : null;
+}
+export function normCountry(country) {
+  if (!country) return null;
+  let c = country?.toLowerCase();
+  if (c === 'uk') c = 'gb';
+  return c.split('_')[0];
+}
+export async function getC0002(optin, akamaiLocale) {
+  const category = 'C0002';
+  let country = normCountry(akamaiLocale || sessionStorage.getItem('akamai'));
+  const explicitConsentCountries = ['gb'];
+  if (optin) {
+    const allowed = {
+      on: true,
+      off: false,
+      unset: country ? !explicitConsentCountries.includes(country) : true,
+    };
+    return { country: normCountry(country), hasC0002: allowed[optin] };
+  }
+
+  const kndctrCookie = getCookie('kndctr_9E1005A551ED61CA0A490D45_AdobeOrg_consent');
+  if (kndctrCookie) {
+    const consent = kndctrCookie.split(';')
+      .reduce((consentByPurpose, categoryPair) => {
+        const [key, value] = categoryPair.split('=');
+        consentByPurpose[key] = value;
+        return consentByPurpose;
+      }, {});
+    if (consent?.general === 'in') return { country, hasC0002: true };
+    if (consent?.general) return { country, hasC0002: false };
+  }
+  const { adobePrivacy } = window;
+  if ((adobePrivacy?.hasUserProvidedConsent() || adobePrivacy?.hasUserProvidedCustomConsent())) {
```

Comment:
This is not needed if you're checking the cookies directly. If you need to wait for the consent signal, this is not the proper way to implement. Details are here: https://wiki.corp.adobe.com/display/adobedotcom/Consent+Management+Platform+Onboarding#ConsentManagementPlatformOnboarding-Usage